@types/navigation.d.ts

```diff
@@ -1,8 +1,14 @@
 import { NavigatorScreenParams } from '@react-navigation/native';
 import { StackScreenProps } from '@react-navigation/stack';
 
+export type AuthenticatedParamsList = {
+  Dashboard: undefined;
```

Comment:
We should follow JS conventions i.e. `d` in `Dashboard` should be lowercase.

The above is also not a good typedef. Typically, undefined is often represented simply via `?`

This feedback is applicable for other typedefs introduced in this PR

Reply:
Regarding the use of undefined, it's intentional to indicate that the route doesn't have params. This aligns with the official React Navigation documentation where undefined is explicitly used to represent routes without parameters, ensuring clarity in the type definitions.

As for the camelCase naming convention, I agree with the need to follow consistent naming patterns. However, the use of camelCase for route names (phoneAuth, otpVerify) and PascalCase for component names (Dashboard, Startup) is a common convention in JS/TS. This helps differentiate between different types of identifiers at a glance. If there's a preferred convention or if there's a rationale for changing this pattern, please let me know.

Reference: [React Navigation Doc](https://reactnavigation.org/docs/typescript/)